docs/06-configuration.md

```diff
@@ -115,6 +115,33 @@ export default ({projectDir}) => {
 
 Note that the final configuration must not be a promise.
 
+## Alternative configuration files
+
+The [CLI] lets you specify a specific configuration file, using the `--config` flag. This file is processed just like an `ava.config.js` file would be. When the `--config` flag is set, the provided file will override configuration from the `package.json` and `ava.config.js` files.
```

Comment:
What happens when I pass a config flag `npx ava --config integration-tests.config.js` + a ava config in the package.json. Are values inherited or they will be treated as two separate configs? This is not clear to me.

Reply:
The provided file (`integration-tests.config.js`) will override the configuration from the `package.json` file.

Reply:
What happens when integration-tests.config.js doesn't override a default but the config in package.json? Are values inherited from the package.json?

Reply:
Could you clarify what you mean by "overriding a default"?

All configuration is overridden. Would it help if I change the line to say:

> the provided file will override all configuration from the `package.json` and `ava.config.js` files.

Reply:
I got it but I will try explain the other scenario.

1. I have a custom config file like `integration.config.js` with zero configuration.
2. I have a `ava` key in my package.json which defines `failFast: true`
3. I will run the tests with `ava --config integration.config.js`
4. Is `failFast` true?

Reply:
No.

Normally, if you had config like that *and* an `ava.config.js` file we'd throw an error. The configuration is not merged.

The `--config` flag forgoes the error but the configuration is still not merged. At this stage, the idea is that you manually extend the configuration. We can explore what merging looks like separately.

Reply:
I like it! Automatic merging could lead to crazy side-effects. Passing multiple configs would be a great option to realize merging e.g if you have a base config and a integration, unit config.

```
--config base.config.js --config int.config.js
```